kolibri/core/public/test/test_api.py

```diff
@@ -557,3 +558,136 @@ def test_update_full_queue_should_scale_keep_alive(self):
         data = response.json()
         self.assertEqual(response.status_code, 200)
         self.assertEqual(data["keep_alive"], 3 * HANDSHAKING_TIME)
+
+    def test_create_soft_deleted_user_should_sync(self):
+        soft_deleted_user = FacilityUser.objects.create(
+            username="soft_deleted",
+            password="***",
+            facility=self.facility,
+        )
+        controller = MorangoProfileController(FacilityUser.morango_profile)
+        controller.serialize_into_store()
+        # Soft delete the user by setting date_deleted
+        soft_deleted_user.date_deleted = timezone.now()
+        soft_deleted_user.save()
+
+        response = self.client.post(
+            reverse("kolibri:core:syncqueue"),
+            data={
+                "user": soft_deleted_user.id,
+                "instance": self.instance_id,
+            },
+            format="json",
+        )
+        data = response.json()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(data["status"], SyncQueueStatus.Ready)
+
+    @mock.patch("kolibri.core.device.models.UserSyncStatus.update_status")
+    def test_create_deleted_user_should_sync(self, mock_update):
+        deleted_user = FacilityUser.objects.create(
+            username="deleted",
+            password="***",
+            facility=self.facility,
+        )
+        user_id = deleted_user.id
+        controller = MorangoProfileController(FacilityUser.morango_profile)
+        controller.serialize_into_store()
```

Comment:
Since LOD syncing uses a partition filter, these tests aren't quite representative of LOD syncing, because this line here is a full-facility serialization. So [with respect to my comment on slack](https://learningequality.slack.com/archives/CB37UM23A/p1758724621284959?thread_ts=1758670722.392489&cid=CB37UM23A), I would update these to use the user's partition filter. The only problematic test that should fail, without the fix described on slack, should be the hard deleted one.